src/controllers/dlgprefcontroller.cpp

```diff
@@ -823,10 +839,37 @@ void DlgPrefController::slotShowMapping(std::shared_ptr<LegacyControllerMapping>
     m_ui.labelLoadedMappingSupportLinks->setText(mappingSupportLinks(pMapping));
     m_ui.labelLoadedMappingScriptFileLinks->setText(mappingFileLinks(pMapping));
 
-    // We mutate this mapping so keep a reference to it while we are using it.
-    // TODO(rryan): Clone it? Technically a waste since nothing else uses this
-    // copy but if someone did they might not expect it to change.
-    m_pMapping = pMapping;
+    if (pMapping) {
+        pMapping->loadSettings(m_pConfig, m_pController->getName());
+        auto settings = pMapping->getSettings();
+        auto* pLayout = pMapping->getSettingsLayout();
+
+        qDeleteAll(m_ui.groupBoxSettings->findChildren<QWidget*>("", Qt::FindDirectChildrenOnly));
```

Comment:
We leak here the implicit created QLayoutItems see:
https://doc.qt.io/qt-6/qlayout.html#addItem 
Also a workaround for a temporary QList

```suggestion
        QLayoutItem *pItem; 
        while ((pItem = m_ui.groupBoxSettings->layout()->takeAt(0)) != nullptr) {
            delete pItem->widget();
            delete pItem;
        }  
```